sys/log/full/src/log_fcb.c

```diff
@@ -699,18 +701,24 @@ log_fcb_copy_entry(struct log *log, struct fcb_entry *entry,
                    struct fcb *dst_fcb)
 {
     struct log_entry_hdr ueh;
-    char data[LOG_PRINTF_MAX_ENTRY_LEN + sizeof(ueh)];
+    char data[LOG_PRINTF_MAX_ENTRY_LEN + LOG_BASE_ENTRY_HDR_SIZE + LOG_IMG_HASHLEN];
+    uint16_t ext_len = 0;
     int dlen;
     int rc;
     struct fcb *fcb_tmp;
 
-    rc = log_fcb_read(log, entry, &ueh, 0, sizeof(ueh));
-    if (rc != sizeof(ueh)) {
+    rc = log_fcb_read(log, entry, &ueh, 0, LOG_BASE_ENTRY_HDR_SIZE);
+    if (rc != LOG_BASE_ENTRY_HDR_SIZE) {
         goto err;
     }
 
-    dlen = min(entry->fe_data_len, LOG_PRINTF_MAX_ENTRY_LEN + sizeof(ueh));
+    if (ueh.ue_flags & LOG_FLAGS_IMG_HASH) {
+        ext_len = LOG_IMG_HASHLEN;
+    }
```

Comment:
Probably better to use `hdr_len = log_hdr_len(&ueh);`.  Then this code doesn't have to get updated if we add a new optional field to the header.